sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedStartFromETagAndFeedRangeImpl.java

```diff
@@ -67,14 +67,17 @@ public int hashCode() {
     }
 
     @Override
-    public void populateRequest(RxDocumentServiceRequest request) {
+    public void populateRequest(RxDocumentServiceRequest request, ChangeFeedMode changeFeedMode) {
         checkNotNull(request, "Argument 'request' must not be null.");
 
         if (this.eTag != null) {
             // On REST level, change feed is using IfNoneMatch/ETag instead of continuation
-            request.getHeaders().put(
-                HttpConstants.HttpHeaders.IF_NONE_MATCH,
-                this.eTag);
+            request.getHeaders().put(HttpConstants.HttpHeaders.IF_NONE_MATCH, this.eTag);
+        } else if (ChangeFeedMode.FULL_FIDELITY == changeFeedMode) {
+            //  If there is no continuation token, we start from now (which is by default).
+            //  On REST level, change feed is using IfNoneMatch/ETag instead of continuation.
+            request.getHeaders().put(HttpConstants.HttpHeaders.IF_NONE_MATCH,
+                HttpConstants.HeaderValues.IF_NONE_MATCH_ALL);
```

Comment:
why we do not need to populate here for Incremental mode?

Reply:
For incremental, this header is not mandatory, and if not present it is default by beginning. For full fidelity, this header is mandatory.